eyelid-match-ops/src/primitives/yashe.rs

```diff
@@ -278,4 +346,116 @@ where
         res.truncate_to_canonical_form();
         res
     }
+
+    pub fn plaintext_add(&self, m1: Message<C>, m2: Message<C>) -> Message<C> {
+        let mut res = m1.m + m2.m;
+        for coeff in res.coeffs_mut() {
+            let mut coeff_res = C::coeff_as_u128(*coeff);
+            coeff_res %= C::t_as_u128();
+            *coeff = coeff_res.into();
+        }
+        res.truncate_to_canonical_form();
+        Message { m: res }
+    }
+
+    pub fn plaintext_mul(&self, m1: Message<C>, m2: Message<C>) -> Message<C> {
+        let mut res = m1.m * m2.m;
+        for coeff in res.coeffs_mut() {
+            let mut coeff_res: i128 = C::coeff_as_u128(*coeff) as i128;
```

Comment:
Rather than casting to `i128` all the time, it would be clearer to add utility methods to the config trait for `i128`.

It should also give us the same or better performance (due to less code duplication).

Reply:
We fixed this.